e2e/scenario_test.go

```diff
@@ -633,6 +633,9 @@ func Test_Ubuntu2204Gen2_ContainerdAirgappedK8sNotCached(t *testing.T) {
 			Cluster: ClusterKubenetAirgap,
 			VHD:     config.VHDUbuntu2204Gen2ContainerdAirgappedK8sNotCached,
 			BootstrapConfigMutator: func(nbc *datamodel.NodeBootstrappingConfiguration) {
+				if nbc.KubeletConfig != nil {
+					delete(nbc.KubeletConfig, "--azure-container-registry-config")
```

Comment:
this flag was removed in AKS 1.31, shouldnt we use that condition to decide when not to configure it.